File: Linear/Dynamic/Linked List/Singly Linked List/01_Traversing/Iterative/Traversing.java
// A Linked List Node
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class IterativeTraversingList {
    // Function to traverse and print the singly linked list
    public static void traverseList(Node head) {
        while (head != null) {
            System.out.print(head.data);
            if (head.next != null)
            System.out.print(" -> ");
            head = head.next;
        }

        System.out.println();
    }

    public static void main(String[] args) {
        // Create a hard-coded linked list:
        // 10 -> 20 -> 30 -> 40
        Node head = new Node(10);
        head.next = new Node(20);
        head.next.next = new Node(30);
        head.next.next.next = new Node(40);

        traverseList(head);
    }
}

File: Linear/Dynamic/Linked List/Doubly Linked List/01_Traversing/Backward/Iterative/Traversing.js
class Node {
    constructor(data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

/*Function to traverse the doubly
linked list in backward direction*/
function backwardTraversal(tail) {
    while (tail !== null) {
        process.stdout.write(tail.data.toString())
        if (tail.prev !== null)
            process.stdout.write(" <-> ")
        tail = tail.prev
    }
    console.log()
}

// Create a hardcoded doubly linked list:
// 1 <-> 2 <-> 3
let head = new Node(1);
let second = new Node(2);
let third = new Node(3);

head.next = second;
second.prev = head;
second.next = third;
third.prev = second;

console.log("Backward Traversal: ")
backwardTraversal(third)